src/ovirtapi/transform.js

```diff
@@ -327,9 +327,9 @@ const VmStatistics = {
         stat.values.value &&
         (stat.name === 'disks.usage'
           ? stat.values.value[0] && stat.values.value[0].detail
-          : stat.values.value.length === 1
-            ? stat.values.value[0].datum
-            : stat.values.value.map(value => value.datum))
+          : (Array.isArray(stat.values.value)
+            ? stat.values.value.map(value => value.datum)
+            : [stat.values.value]))
```

Comment:
Sorry, I think I've misunderstood the comment (but let's @sjd78 clarify that):
> no values -> undefined, 1 value -> value.datum, >1 values -> [...values.datum]

The left side is input(JSON from backend) and right is our output, right? What are the possible values of the input: `undefined` or an array? Or is it  sometimes a single object (not wrapped in array)?

Anyway, the current version could result in `[undefined]`. We should probably return empty array then.

Reply:
Based on backend data, this is always returned as an array (List<Integer>), so not sure converting  here is required.

I think that the fix for stat.name !== 'disks.usage' should be the same as for 'disks.usage'. i.e.: checking that  stat.values.value[i]  is defined because in case of null values, the datum was not set on backend.
@sjd78 ?

Reply:
@rszwajko @sgratch I checked that and when `stat.values.value` is `undefined`, we end up with `datum` set to `undefined`, not `[undefined]` because of [this](https://github.com/oVirt/ovirt-web-ui/pull/1358/files#diff-83343b99424fb2a62ce7723a77810ebe7d4f3ccf8fd53ad4cd92c244902dc80aR327). And if `stat.values.value` is something but not an array, it will become wrapped in array because of [this change](https://github.com/oVirt/ovirt-web-ui/pull/1358/files#diff-83343b99424fb2a62ce7723a77810ebe7d4f3ccf8fd53ad4cd92c244902dc80aR332). IMO this change is okay.

Reply:
I'm gonna suggest just rewriting this section to be less cryptic.

How does this look:
```js
    for (const stat: ApiVmStatisticType of statistics) {
      if (stat.name === 'elapsed.time') {
        base.elapsedUptime.datum = stat.values.value[0].datum
        base.elapsedUptime.description = stat.description
      }

      if (stat.kind !== 'gauge') continue

      // no values -> undefined
      let datum: any

      if (stat.values && stat.values.value) {
        if (stat.type === 'decimal' || stat.type === 'integer') {
          // 1 value -> value.datum, >1 values -> [...values.datum]
          datum = Array.isArray(stat.values.value) && stat.values.value.length === 1
            ? stat.values.value[0].datum
            : stat.values.value.map(value => value.datum)
        }

        if (stat.type === 'string') {
          // 1 value -> value.detail, >1 values -> [...values.detail]
          datum = Array.isArray(stat.values.value) && stat.values.value.length === 1
            ? stat.values.value[0].detail
            : stat.values.value.map(value => value.detail)
        }
      }

      // further parse the JSON string in `disk.usage` stats
      if (stat.name === 'disks.usage' && !!datum) {
        datum = JSON.parse(datum).map(data => {
          data.total = convertInt(data.total)
          data.used = convertInt(data.used)
          return data
        })
      }

      const nameParts = /^(memory|cpu|network|disks)\.(.*)?$/.exec(stat.name)
      if (nameParts) {
        base[nameParts[1]][nameParts[2]] = {
          datum,
          unit: stat.unit,
          description: stat.description,
        }
      }
    }
```

And then the other users of the stats data can go back to how they were.